preprocess_segment.sh

```diff
@@ -99,33 +59,51 @@ sct_check_dependencies -short
 
 # Go to folder where data will be copied and processed
 cd $PATH_DATA_PROCESSED
+
 # Copy source images
-rsync -avzh $PATH_DATA/$SUBJECT .
+rsync -avzh $PATH_DATA/$SUBJECT/$DATA_TYPE/* .
 
-# T2w
+
+# Segment spinal cord (SC) if does not exist
 # ======================================================================================================================
-cd "${SUBJECT}/anat/"
-file_t2="${SUBJECT}_${SUFFIX_T2w}"
-# Segment spinal cord 
-segment_if_does_not_exist "${file_t2}" "t2"
-file_t2_seg="${FILESEG}"
-# Create vertebral labels 
-label_if_does_not_exist "${file_t2}" "${file_t2_seg}"
-
-# T1w
+
+FILE="${SUBJECT}${IMAGE_SUFFIX}.nii.gz"
+FILESEG="${SUBJECT}${IMAGE_SUFFIX}_label-SC_seg.nii.gz"
+
+echo "Looking for segmentation: ${FILESEG}"
+if [[ -e ${FILESEG} ]]; then
+  echo "Found! Using SC segmentation that exists."
+  sct_qc -i ${FILE} -s ${FILESEG} -p sct_deepseg_sc -qc ${PATH_QC} -qc-subject ${SUBJECT}
+else
+  echo "Not found. Proceeding with automatic segmentation."
+  # Segment spinal cord
+  sct_deepseg_sc -i ${FILE} -o ${FILESEG} -c ${CONTRAST} -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  # TODO: MOVE THAT FILE UNDER derivatives/labels
```

Comment:
@NadiaBlostein after thinking more about it, I think we should _not_ move the outputs in the derivatives folder of the dataset automatically. The reason is that, if someone tries running the script, it will overwrite the data in the derivatives, which will be an annoyance.

Instead, we should do this move while doing the manual correction of the segmentations/labels. In fact, this is what is currently done by other projects. Also see https://github.com/neuropoly/template/issues/58

Reply:
Lines 73 and 89 of preprocess_segment.py check if the files exist to avoid overwriting anything. This also allows the straightening.cache, straight_ref.nii.gz and warp*.nii.gz files for each subject to be saved separately. 

However, if you prefer us to remain consistent with what everyone else does, I’ll take a look through the other projects to find a repository that I could use as a “template” for this.

Reply:
preprocess_segment.py is not used anymore, is it? I’m not sure I understand your comment.

Also: straightening.cache etc. have nothing to do with my comment because these files were not copied anyway. I think there is a misunderstanding that will
likely be better resolved in a meeting.

Reply:
> My apologies, I made a typo (was on phone); was referring to preprocess_segment.sh. Will correct comment now.

👍 

> I understood that you may have at least wanted the subject-specific warp files saved. We'll chat next week.

No. What I meant in the meeting is that we want these subject-specific files in their own folders (as opposed to a flat directory where there could be file conflicts)-- I did not mean that these data should be ultimately stored in the git-annexed source data